### PR TITLE
Make ads131m driver agnostic

### LINKED ISSUE #155: [Driver] - Make ads131m driver platform agnostic
### Current Features

this driver depends on stm32 HAL

### Desired Additional Features

We want these to work with all potential hardware. This involves replacing HAL specific code with generalized code that utilized function pointers to use HAL functionality. The way in which we want to handle this is outlined [in this doc](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/430997517/C+Coding+Standards) under the "Driver Generalization" header. 

Also a part of this ticket, find all references to this driver in Shepherd and Cerberus, and fix them to accommodate the new driver structure. A good way to check this is to update your embedded-base submodule to reflect the changes made to this driver, and then try building shepherd and cerberus. If they relied on this device, they will likely fail to build unless you make the appropriate changes

### Screenshots (as needed)

_No response_

### PR DESCRIPTION
**Pull request by bjackson312006**
## Changes

Removed HAL-specific code in ads131m04 driver and added function pointers

Closes #155

### CODE REVIEW

File: general/include/ads131m04.h [hunk 1]
```diff
@@ -3,18 +3,25 @@
 
 #include "stm32xx_hal.h"
 
+/* Function Pointers*/
+typedef void (*GPIO_WriteFuncPtr)(uint8_t cs_pin, uint8_t PinState);
```
Comment (dyldonahue) [inline general/include/ads131m04.h:7]: small thing - but cs pin is actually an SPI thing. Its used in the GPIO functions below because its just a GPIO pin that needs to be pulled low to write SPI, and otherwise pulled high. 

TLDR; the variable here should just be "pin" instead of "cs_pin" - because the gpio could write to any pin not just SPis CS pin

File: general/include/ads131m04.h [hunk 2]
```diff
@@ -3,18 +3,25 @@
...
+typedef void (*SPI_TransmitFuncPtr)(uint8_t *pData, uint16_t size);
```
Comment (dyldonahue) [inline general/include/ads131m04.h:8]: nitpick but can the names change like this: 

SPI_TransimitFuncPtr --> "WritePtr"
GPIO_WriteFunctPtr --> GPIO_WritePtr
SPI_ReceiveFuncPtr -->ReadPtr

local_GPIO_Write --> gpio_write
local_SPI_Transmit --> write
local_SPI_Recieve --> read

**Review by dyldonahue — CHANGES_REQUESTED** [review]
looks good, a few naming changes and this should be good to go

**Review by dyldonahue — APPROVED** [review]
